src/models/appmodel.ts

```diff
@@ -13,7 +13,23 @@ export class AppModel implements IAppModel {
     get selectedFlavorIndex(): number { return this._selectedFlavorIndex; }
     set selectedFlavorIndex(value: number) { this._selectedFlavorIndex = value; }
 
+    // we can create a dedicated Class for SelectedItem like: class SelectedItem etc. but it will be too complicated
+    @observable private _selectedItem = { label: "", value: "" };
```

Comment:
_selectedItem must be the same type as the items source array.

Reply:
I see. WIll extract the value string from the Select object.

As the ReactSelect passes the label and value object on Select and we need to map string[] before passing options to the ReactSelect and the same when we are selecting item and making it _selectedItem value, I think, in this case we can create some mapper regarding this reason.

Reply:
Done, added a simple mapper